metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -238,7 +241,7 @@ func NewResourceMetadataEnricher(
 			case *kubernetes.StatefulSet:
 				m[id] = metaGen.Generate(StatefulSetResource, r)
 			case *kubernetes.Namespace:
-				m[id] = metaGen.Generate("namespace", r)
+				m[id] = metaGen.Generate(NamespaceResource, r)
```

Comment:
Why not to use the metadata generator that is already initialized a few lines above? We have a specific Namespace metadata generator to use instead of using the generic Resource one. The same we do for Pods, Services and Nodes:

```suggestion
				m[id] = namespaceMeta.Generate(r)
```

I think this is more accurate and straight-forward.

Reply:
Good suggestion, I tried and it worked. The comment above is no longer true then, I will try to remove the field in `kubernetes.namespace.name`.